src/functionalTest/java/uk/gov/hmcts/dts/fact/CourtsEndpointTest.java

```diff
@@ -107,6 +109,38 @@ public void shouldRetrieveCourtDetailBySlug() {
         assertThat(court.getSlug()).isEqualTo(BIRMINGHAM_CIVIL_AND_FAMILY_JUSTICE_CENTRE);
     }
 
+    @Test
+    public void shouldRetrieveCourtsByPrefixWhereDisplayedFalseAndCaseLower() {
+        final var response = doGetRequest(COURT_SEARCH_BY_PREFIX_AND_ACTIVE_ENDPOINT + "?prefix=a&active=false");
+        assertThat(response.statusCode()).isEqualTo(OK.value());
+
+        final List<CourtReference> courtReferences = response.body().jsonPath().getList(".", CourtReference.class);
+        Assertions.assertTrue(courtReferences.stream().allMatch(c -> c.getName().charAt(0) == 'A'));
+        Assertions.assertTrue(courtReferences.stream().allMatch(c -> c.getSlug().charAt(0) == 'a'));
+    }
+
+    @Test
+    public void shouldRetrieveCourtsByPrefixWhereDispayedTrueAndCaseUpper() {
+        final var response = doGetRequest(COURT_SEARCH_BY_PREFIX_AND_ACTIVE_ENDPOINT + "?prefix=B&active=true");
+        assertThat(response.statusCode()).isEqualTo(OK.value());
+
+        final List<CourtReference> courtReferences = response.body().jsonPath().getList(".", CourtReference.class);
+        Assertions.assertTrue(courtReferences.stream().allMatch(c -> c.getName().charAt(0) == 'B'));
+        Assertions.assertTrue(courtReferences.stream().allMatch(c -> c.getSlug().charAt(0) == 'b'));
+    }
+
+    @Test
+    public void shouldReturnAnErrorWhenSizeConstraintBreached() {
+        final var response = doGetRequest(COURT_SEARCH_BY_PREFIX_AND_ACTIVE_ENDPOINT + "?prefix=mosh&active=true");
+        assertThat(response.statusCode()).isEqualTo(INTERNAL_SERVER_ERROR.value());
```

Comment:
IS that possible to get the endpoint to return a bad request rather than internal server error?

Reply:
It's possible, but we would need to remove the @size constraint from the endpoint, and do a manual check ourselves instead and then throw our own exception

I.e change from @size constrant to

if (prefix size != 1) 
then throw bad request exception